src/command_modules/azure-cli-role/azure/cli/command_modules/role/commands.py

```diff
@@ -80,7 +82,9 @@ def load_command_table(self, _):
         g.custom_command('delete', 'delete_application')
         g.custom_command('list', 'list_apps')
         g.custom_command('show', 'show_application', exception_handler=empty_on_404)
-        g.custom_command('update', 'update_application')
+        g.generic_update_command('update', setter_name='patch_application', setter_type=role_custom,
+                                 getter_name='show_application', getter_type=role_custom,
```

Comment:
The purpose of using the custom getter is to take other kinds of user friendly application identities (app id uri) instead of the guid id required the API/get. Since the instance returned from the getter is not used, so the getter is being used like a validator. I have tried to remove the getter parameters from the command authoring, but turned out I need to update the core layer pretty extensively, as the generic update support has a very strong assumption on the getter being provided.